### DISCUSSION TITLE
Get a string value for buildDetails.reason instead of the enum integer value

### DISCUSSION THREAD

**Discussion by raydixon** [rfennell/AzurePipelines#1312] [Q&A]
When I use the `buildDetails.reason` field, it always returns the integer value of the enumeration. I found the definitions for the values in the [BuildReason enum](https://docs.microsoft.com/en-us/javascript/api/azure-devops-extension-api/buildreason) documentation.

I would like a way to get the string values for the `reason` field, like `Manual`, `BatchedCI`, `PullRequest`, etc., instead of `1`, `4`, `256`, etc. Other than using several comparison statements (what I'm doing, now), is there a way to get a reader-friendly value for the build reason rather than an integer?

**Comment by rfennell** [comment]
I would suggest a custom Handlebars extension that wrapper a switch statement.

You can find [sample of other extensions here](https://github.com/rfennell/AzurePipelines/tree/main/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203)

If you write this extension, feel free to submit it as a PR so I can add it to sample library.
